packages/create-cloudflare/templates-experimental/react/c3.ts

```diff
@@ -0,0 +1,189 @@
+			// Add an import of the @cloudflare/vite-plugin
+			// ```
+			// import {cloudflare} from "@cloudflare/vite-plugin";
+			// ```
+			const lastImportIndex = n.node.body.findLastIndex(
+				(t) => t.type === "ImportDeclaration",
+			);
+			const lastImport = n.get("body", lastImportIndex);
+			const importAst = b.importDeclaration(
+				[b.importSpecifier(b.identifier("cloudflare"))],
+				b.stringLiteral("@cloudflare/vite-plugin"),
+			);
+			lastImport.insertAfter(importAst);
+
+			return this.traverse(n);
+		},
+		visitCallExpression: function (n) {
+			// Add the imported plugin to the config
+			// ```
+			// defineConfig({
+			//   plugins: [react(), cloudflare()],
+			// });
+			const callee = n.node.callee as types.namedTypes.Identifier;
+			if (callee.name !== "defineConfig") {
+				return this.traverse(n);
+			}
+
+			const config = n.node.arguments[0] as types.namedTypes.ObjectExpression;
+
+			const pluginsProp = config.properties.find(
+				(prop) =>
+					prop.type === "ObjectProperty" && getPropertyName(prop) === "plugins",
+			) as types.namedTypes.ObjectProperty;
+			const pluginsArray =
+				pluginsProp.value as types.namedTypes.ArrayExpression;
+
+			mergeObjectProperties(
+				n.node.arguments[0] as types.namedTypes.ObjectExpression,
+				[
+					b.objectProperty(
+						b.identifier("plugins"),
+						b.arrayExpression([
+							...pluginsArray.elements,
+							b.callExpression(b.identifier("cloudflare"), []),
+						]),
+					),
+				],
+			);
+
+			return false;
+		},
+	});
+}
+
+function updateTsconfigJson() {
+	const s = spinner();
+	s.start(`Updating tsconfig.json config`);
+	// Add a reference to the extra tsconfig.worker.json file.
+	// ```
+	// "references": [ ..., { path: "./tsconfig.worker.json" } ]
+	// ```
+	const tsconfig = readJSON("tsconfig.json") as { references: object[] };
+	if (tsconfig && typeof tsconfig === "object") {
+		tsconfig.references ??= [];
+		tsconfig.references.push({ path: "./tsconfig.worker.json" });
+	}
+	writeJSON("tsconfig.json", tsconfig);
+	s.stop(`${brandColor(`updated`)} ${dim(`\`angular.json\``)}`);
+}
+
+async function getVariant() {
+	const variantsOptions = [
+		{
+			value: "react-ts",
+			lang: "ts",
+			label: "TypeScript",
+		},
+		{
+			value: "react-swc-ts",
+			lang: "ts",
+			label: "TypeScript + SWC",
+		},
+		{
+			value: "react",
+			lang: "js",
+			label: "JavaScript",
+		},
+		{
+			value: "react-swc",
+			lang: "js",
+			label: "JavaScript + SWC",
+		},
+	];
+	const value = await inputPrompt({
+		type: "select",
+		question: "Select a variant:",
+		label: "variant",
+		options: variantsOptions,
+		defaultValue: variantsOptions[0].value,
+	});
+
+	const selected = variantsOptions.find((variant) => variant.value === value);
+	assert(selected, "Expected a variant to be selected");
+	return selected;
+}
+
+const config: TemplateConfig = {
+	configVersion: 1,
+	id: "react",
+	frameworkCli: "create-vite",
+	displayName: "React",
+	platform: "workers",
```

Comment:
```suggestion
	platform: "Workers",
```
?

Reply:
'fraid not. This is a discriminator, not user facing